requirements.txt

```diff
@@ -1,7 +1,7 @@
 ninja
 flaky
 numpy
-pennylane>=0.15
+git+https://github.com/PennyLaneAI/pennylane.git@master
```

Comment:
I am not so sure whether this will yield the behavior we want. This means `pip install pennylane-lightning` will install the `master` version (after removing the installed PennyLane).

Reply:
I think this is what we want, as the only use-case for this docker image right now is to work with the latest Pennylane, and using the master branch of lightning with the non master of PennyLane is more likely to cause issues than both at the same tag. We will likely deprecate this builder in favour of the PL Dockerfiles in a release or two, as it seems to provide a better end-to-end ecosystem support.

Reply:
Oh no, you are right, this was not a patch for the Dockerfile. Ignore my previous comment. @chaeyeunpark is 100% correct.

Reply:
Can we move this into the Dockerfile and directly patch the requirements.txt from there instead of making the change outright?

Reply:
Hi @chaeyeunpark, this change yielded exactly the change I wanted. 
After some discussions with @mlxd, we decided to move it to a new requirements-dev.txt, though.
Thank you all for the quick reviews.